slatedb/src/retrying_object_store.rs

```diff
@@ -131,17 +135,74 @@ impl ObjectStore for RetryingObjectStore {
     }
 
     fn list(&self, prefix: Option<&Path>) -> BoxStream<'static, object_store::Result<ObjectMeta>> {
-        // Delegates directly; listing returns a stream so per-page retry is out of scope here.
-        self.inner.list(prefix)
+        let inner = Arc::clone(&self.inner);
```

Comment:
This issue was causing failures in the chaos test, so I am fixing it as part of the PR.